### PR TITLE
Parse dates with multiple formats

### PR DESCRIPTION
**Pull request by tgvLt**
[baeldung-articles](https://jira.baeldung.com/browse/BAEL)[BAEL-5770](https://jira.baeldung.com/browse/BAEL-5770)

### CODE REVIEW

**Review by tgvLt — COMMENTED** [review]
Remove regex

File: core-java-modules/core-java-8-datetime-2/src/test/java/com/baeldung/parsingDates/SimpleParseDateTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package com.baeldung.parsingDates;
+
+import com.baeldung.parsingDates.SimpleDateTimeFormat;
+import com.baeldung.parsingDates.SimpleDateTimeFormater;
+import com.baeldung.parsingDates.SimpleDateUtils;
+import com.baeldung.parsingDates.SimpleParseDate;
+import java.util.Arrays;
+import org.junit.*;
+
+public class SimpleParseDateTest {
```
Comment (Bwvolleyball) [inline core-java-modules/core-java-8-datetime-2/src/test/java/com/baeldung/parsingDates/SimpleParseDateTest.java:10]: Looks like this file name is causing the build to fail: https://jenkins.baeldung.com/job/tutorials-build-pr/15338/console

I think we need to rename it.

```suggestion
public class SimpleParseDateUnitTest {

```

File: core-java-modules/core-java-8-datetime-2/src/test/java/com/baeldung/parsingDates/SimpleParseDateUnitTest.java
Comment (KevinGilmore) [inline core-java-modules/core-java-8-datetime-2/src/test/java/com/baeldung/parsingDates/SimpleParseDateUnitTest.java:1]: Let's rename the test methods according to the guidelines, using the "givenX_whenY_thenZ" pattern. Note that the "givenX_" part is optional.

Also, the order of arguments to assertEquals should be swapped: assertEquals(expected, actual).

**Review by KevinGilmore — COMMENTED** [review]
See comments on the unit test file

File: core-java-modules/core-java-8-datetime-2/src/main/java/com/baeldung/parsingDates/SimpleDateTimeFormater.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+    public static String parseDate(String date) {
+        DateTimeFormatterBuilder dateTimeFormatterBuilder = new DateTimeFormatterBuilder()
+                .append(DateTimeFormatter.ofPattern("[MM/dd/yyyy]" + "[dd-MM-yyyy]" + "[yyyy-MM-dd]"));
+
+        DateTimeFormatter dateTimeFormatter = dateTimeFormatterBuilder.toFormatter();
+
+        return LocalDate.parse(date, dateTimeFormatter).toString();
+    }
+}
```
Comment (Bwvolleyball) [inline core-java-modules/core-java-8-datetime-2/src/main/java/com/baeldung/parsingDates/SimpleDateTimeFormater.java:17]: It looks like we are turning this into a string without needing too.

File: core-java-modules/core-java-8-datetime-2/src/main/java/com/baeldung/parsingDates/SimpleDateTimeFormater.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package com.baeldung.parsingDates;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+
+public class SimpleDateTimeFormater {
+
+    public static String parseDate(String date) {
```
Comment (Bwvolleyball) [inline core-java-modules/core-java-8-datetime-2/src/main/java/com/baeldung/parsingDates/SimpleDateTimeFormater.java:9]:
```suggestion
    public static LocalDate parseDate(String date) {

```

File: core-java-modules/core-java-8-datetime-2/src/main/java/com/baeldung/parsingDates/SimpleDateUtils.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package com.baeldung.parsingDates;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import org.apache.commons.lang3.time.DateUtils;
+
+public class SimpleDateUtils {
+
+    public static String parseDate(String date) {
```
Comment (Bwvolleyball) [inline core-java-modules/core-java-8-datetime-2/src/main/java/com/baeldung/parsingDates/SimpleDateUtils.java:10]:
```suggestion
    public static Date parseDate(String date) {

```

File: core-java-modules/core-java-8-datetime-2/src/main/java/com/baeldung/parsingDates/SimpleDateUtils.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import org.apache.commons.lang3.time.DateUtils;
+
+public class SimpleDateUtils {
+
+    public static String parseDate(String date) {
+
+        DateFormat df = new SimpleDateFormat("dd/MM/yyyy");
```
Comment (Bwvolleyball) [inline core-java-modules/core-java-8-datetime-2/src/main/java/com/baeldung/parsingDates/SimpleDateUtils.java:12]:
```suggestion
```

File: core-java-modules/core-java-8-datetime-2/src/main/java/com/baeldung/parsingDates/SimpleDateUtils.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+        try {
+            return df.format(DateUtils.parseDateStrictly(date,
+                    new String[]{"yyyy/MM/dd", "dd/MM/yyyy", "yyyy-MM-dd"}));
```
Comment (Bwvolleyball) [inline core-java-modules/core-java-8-datetime-2/src/main/java/com/baeldung/parsingDates/SimpleDateUtils.java:15]:
```suggestion
            return DateUtils.parseDateStrictly(date,
                    new String[]{"yyyy/MM/dd", "dd/MM/yyyy", "yyyy-MM-dd"});
```

**Pull request merged**